### PR TITLE
[Term Entry] Discrete Math: Proof by Strong Induction

### PR DESCRIPTION
**Pull request by sreeharsha-rav**
### Description

- Created a term entry for proof by strong induction.
- Closes Issue #2393 

### Type of Change

- Added description about proof by strong induction
- Described steps for performing proof by strong induction
- Illustrated an example for proof by strong induction
- Included a Python codebyte to implement given example

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by sreeharsha-rav** [comment]
@KTom101 I made the pull request, is there anything to be done on my side?

File: content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+## Codebyte Example
+
+```codebyte/python
+x = 3
+s = 2*(x-1) - (x-2)
+
+print(x == s)
+```
```
Comment (SSwiniarski) [inline content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md:63]:
```suggestion
```
I don't think the codebyte is adding anything to this entry.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@sreeharsha-rav Sorry for the delay. I did a first review and left a few comments.

**Comment by sreeharsha-rav** [comment]
@SSwiniarski I committed the requested changes, thanks for your review.

**Review by SSwiniarski — APPROVED** [review]
@sreeharsha-rav looks good for a second review.

File: content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  2. _Induction Hypothesis_: Suppose that the statement is true for some integer `k`, then the statement must be true for `k+1`.
+  3. _Proof_: Prove that the induction hypothesis is true for `k+1`.
+
+## Example
+
+Statement: `2(x-1) - (x-2) = x for x >= 2`
+
+Base Cases:
+
+```
```
Comment (KTom101) [inline content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md:31]:
```suggestion
```plaintext
```

File: content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+At x = 2 , 2(2 - 1) - (2 - 2) = 2(1) - (0) = 2 + 0 = 2 and x = 2. Thus, the statement is true at x = 2.
+
+At x = 3, 2(3 - 1) - (3 - 2) = 2(2) - (1) = 4 - 1 = 3 and x = 3. Thus, the statement is true at x = 3.
+```
+
+Induction Hypothesis:
+
+```
```
Comment (KTom101) [inline content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md:39]:
```suggestion
```plaintext
```

File: content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+Suppose at x = k, 2(k-1) - (k-2) = k
+Then at x = k+1, we need to prove that 2((k+1)-1) - ((k+1)-2) = k+1
+```
+
+Proof:
+
+```
```
Comment (KTom101) [inline content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md:46]:
```suggestion
```plaintext
```

File: content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+2((k+1)-1) - ((k+1)-2) = 2(k+1-1) - (k+1-2) 
+                       = 2(k-1+1) - (k-2+1) 
+                       = 2(k-1) + 2(1) - (k-2) -(+1) 
+                       = 2(k-1) - (k-2) + 2 - 1
+                       = 2(k-1) - (k-2) + 1 
+                       = k + 1  [Substitute from x = k, we know that 2(k-1) - (k-2) = k]
```
Comment (KTom101) [inline content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md:52]:
```suggestion
                       = k + 1  [Substituting from x = k, which states that 2(k-1) - (k-2) = k]
```

File: content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+At x = 2 , 2(2 - 1) - (2 - 2) = 2(1) - (0) = 2 + 0 = 2 and x = 2. Thus, the statement is true at x = 2.
+
+At x = 3, 2(3 - 1) - (3 - 2) = 2(2) - (1) = 4 - 1 = 3 and x = 3. Thus, the statement is true at x = 3.
+```
+
+Induction Hypothesis:
+
+```
+Suppose at x = k, 2(k-1) - (k-2) = k
+Then at x = k+1, we need to prove that 2((k+1)-1) - ((k+1)-2) = k+1
```
Comment (KTom101) [inline content/discrete-math/concepts/proofs/terms/proof-by-strong-induction/proof-by-strong-induction.md:41]: avoid using the pronoun "we"
```suggestion
Then at x = k+1, prove that 2((k+1)-1) - ((k+1)-2) = k+1
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@sreeharsha-rav I added some changes to this entry.

**Comment by sreeharsha-rav** [comment]
@KTom101 I committed the requested changes. Thanks for your review.

**Review by KTom101 — APPROVED** [review]
@sreeharsha-rav Looks good, I just ran `npm run format:all` to fix the failed [checks](https://www.codecademy.com/resources/docs/contribution-guide/getting-set-up-on-your-computer#:~:text=There%20are%20continuous%20integration%20(CI)%20workflows).

**Pull request merged**